### ISSUE TITLE
Disabled TextInput still receives focus

### ISSUE THREAD

**Issue by ubruhin** [slint-ui/slint#8626]
### Bug Description

A TextInput with `enabled: false;` is still part of the focus loop. There is no visual indicator of that (no cursor, no selection) but the `changed has-focus` handler is triggered and the focus is "nowhere" then instead of skipping the TextInput.

In the code below, tabulator key cycles through elements "1", "2", "4" instead of just "1", "4".

In addition, the `enabled` property does not seem to be documented (https://snapshots.slint.dev/master/docs/slint/reference/keyboard-input/textinput/).

### Reproducible Code (if applicable)

```slint
import {Button} from "std-widgets.slint";

export component Demo {
    width: 300px;
    height: self.preferred-height;

    property <string> focused;

    VerticalLayout {
        spacing: 5px;
        padding: 5px;

        TextInput {
            text: "Enabled";
            enabled: true;

            changed has-focus => { 
                if self.has-focus { focused = "1"; }
            }
        }

        TextInput {
            text: "Disabled";
            enabled: false;

            changed has-focus => { 
                if self.has-focus { focused = "2"; }
            }
        }

        Button {
            text: "Disabled";
            enabled: false;

            changed has-focus => { 
                if self.has-focus { focused = "3"; }
            }
        }

        Button {
            text: "Enabled";
            enabled: true;

            changed has-focus => { 
                if self.has-focus { focused = "4"; }
            }
        }

        Text {
            text: "Focus: " + focused;
        }
    }
}
```

### Environment Details

- Slint Version: master
- Platform/OS: Linux
- Programming Language: C++
- Backend/Renderer: Qt, Reproducible in SlintPad
